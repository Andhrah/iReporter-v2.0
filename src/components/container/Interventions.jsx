import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import InterventionsView from '../presentational/Interventions';
import { getAllInterventions } from '../../actions';
import Navbar from '../presentational/Navbar';

export class Interventions extends Component {
  componentDidMount() {
    const {
      getAllInterventions: fetchInterventions,
    } = this.props;
    fetchInterventions();
  }

  render() {
    const { interventions } = this.props;

    return (
      <>
        <Navbar
          firstname={localStorage.getItem('firstname')}
          lastname={localStorage.getItem('lastname')}
          email={localStorage.getItem('email')}
          username={localStorage.getItem('username')}
          />
        <InterventionsView interventions={interventions} />
      </>
    );
  }
}

Interventions.propTypes = {
  getAllInterventions: PropTypes.func,
  interventions: PropTypes.array,
};

Interventions.defaultProps = {
  getAllInterventions: () => {},
  interventions: [],
};

const mapStateToProps = state => ({
  interventions: state.allInterventions.data,
});

export default connect(mapStateToProps, { getAllInterventions })(Interventions);
