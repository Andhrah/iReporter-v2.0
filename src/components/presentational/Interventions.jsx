import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Image from './Image';

const Interventions = ({ interventions }) => (
  <section className="section-flex">
    <div className="jumbotron">
      <div className="title-container">
        <h2>
          Interventions Reports
        </h2>
        <Link to="/redflags">
          <em>See Red-Flag reports</em>
        </Link>
      </div>
    </div>
    <div className="grid-container" id="red-flag-grid-container">
      {
        interventions.map(intervention => (
          <div className="incident-container-all list incidents" key={intervention.id}>
            <Image src="http://lorempixel.com/400/400" alt="red-flag incident" />
            <div id="created-by" className="author">
              <em id="created-on">{`created on ${new Date(intervention.created_on).toDateString()} by`}</em>
              {' '}
              {intervention.created_by}
            </div>
            <div className="location">
              <p>
                Location:
                {' '}
                <span><em>{intervention.location}</em></span>
                {' '}
              </p>
            </div>
            <div className="status">
              <p className="none">
                Status:
                {' '}
                <span><em>{intervention.status}</em></span>
              </p>
            </div>
            <div className="comment">
              <h4>
                Dispalay me
                {/* {intervention.intervention_reasons} */}
              </h4>
              <p>
                {intervention.comment.length <= 100 ? intervention.comment : `${intervention.comment.substring(0, 100)} ...`}
              </p>
            </div>
            <div>
              <Link to="redflag" className="read-more">
                Read more
                {' '}
                <i className="fas fa-angle-right" />
              </Link>
            </div>
          </div>
        ))
      }
    </div>
  </section>
);

Interventions.propTypes = {
  interventions: PropTypes.array,
};


Interventions.defaultProps = {
  interventions: [],
};

export default Interventions;
