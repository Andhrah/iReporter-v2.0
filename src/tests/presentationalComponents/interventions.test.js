import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Interventions from '../../components/presentational/Interventions';

const props = {
  interventions: [{
    id: 9,
    created_on: '2018-12-29T00:00:00.000Z',
    created_by: 2,
    intervention_reasons: [
      'Bribery',
      'emblezzlement'
    ],
    location: '6.605874, 8.34917',
    status: 'Under Investigation',
    images: [
      'image'
    ],
    videos: [
      'video'
    ],
    comment: 'We need government intervention'
  }],
};

describe('Interventions test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Interventions {...props} />
      </MemoryRouter>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
