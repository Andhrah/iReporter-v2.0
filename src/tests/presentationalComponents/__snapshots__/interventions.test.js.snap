// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interventions test suite renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "5ucwr8",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "5ucwr8",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Interventions
      interventions={
        Array [
          Object {
            "comment": "We need government intervention",
            "created_by": 2,
            "created_on": "2018-12-29T00:00:00.000Z",
            "id": 9,
            "images": Array [
              "image",
            ],
            "intervention_reasons": Array [
              "Bribery",
              "emblezzlement",
            ],
            "location": "6.605874, 8.34917",
            "status": "Under Investigation",
            "videos": Array [
              "video",
            ],
          },
        ]
      }
    >
      <section
        className="section-flex"
      >
        <div
          className="jumbotron"
        >
          <div
            className="title-container"
          >
            <h2>
              Interventions Reports
            </h2>
            <Link
              to="/redflags"
            >
              <a
                href="/redflags"
                onClick={[Function]}
              >
                <em>
                  See Red-Flag reports
                </em>
              </a>
            </Link>
          </div>
        </div>
        <div
          className="grid-container"
          id="red-flag-grid-container"
        >
          <div
            className="incident-container-all list incidents"
            key="9"
          >
            <Image
              alt="red-flag incident"
              src="http://lorempixel.com/400/400"
            >
              <img
                alt="red-flag incident"
                className="red-flag-image"
                src="http://lorempixel.com/400/400"
              />
            </Image>
            <div
              className="author"
              id="created-by"
            >
              <em
                id="created-on"
              >
                created on Sat Dec 29 2018 by
              </em>
               
              2
            </div>
            <div
              className="location"
            >
              <p>
                Location:
                 
                <span>
                  <em>
                    6.605874, 8.34917
                  </em>
                </span>
                 
              </p>
            </div>
            <div
              className="status"
            >
              <p
                className="none"
              >
                Status:
                 
                <span>
                  <em>
                    Under Investigation
                  </em>
                </span>
              </p>
            </div>
            <div
              className="comment"
            >
              <h4>
                Dispalay me
              </h4>
              <p>
                We need government intervention
              </p>
            </div>
            <div>
              <Link
                className="read-more"
                to="redflag/9"
              >
                <a
                  className="read-more"
                  href="/redflag/9"
                  onClick={[Function]}
                >
                  Read more
                   
                  <i
                    className="fas fa-angle-right"
                  />
                </a>
              </Link>
            </div>
          </div>
        </div>
      </section>
    </Interventions>
  </Router>
</MemoryRouter>
`;
